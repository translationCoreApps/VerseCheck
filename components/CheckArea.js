import React from 'react'
import {Col} from 'react-bootstrap'

import SelectArea from './SelectArea'
import DirectionsArea from './DirectionsArea'
import EditVerseArea from './EditVerseArea'
import CommentArea from './CommentArea'
import style from '../css/Style';

let CheckArea = (props) => {
  let modeArea
  switch(props.mode) {
    case 'edit':
    modeArea = <EditVerseArea
      tags={props.tags}
      verseText={props.verseText}
      verseChanged={props.verseChanged}
      actions={props.actions}
      dir = {props.projectDetailsReducer.manifest.target_language.direction} />
    break
    case 'comment':
    modeArea = <CommentArea comment={props.comment} actions={props.actions} />
    break
    case 'select':
    modeArea = <DirectionsArea quote={props.contextIdReducer.contextId.quote} />
    break
    default:
    modeArea = <DirectionsArea quote={props.contextIdReducer.contextId.quote} />
  }

  return (
    <div style={style.checkArea}>
      <SelectArea {...props} />
      <div style={{ borderLeft: '1px solid var(--border-color)', flex: 1, display: 'flex' }}>
        {modeArea}
      </div>
    </div>
  )
}

module.exports = CheckArea
