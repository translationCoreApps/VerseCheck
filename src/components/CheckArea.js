import React, { Component } from 'react';
import PropTypes from 'prop-types';
// components
import DefaultArea from './DefaultArea';
import SelectionArea from './SelectionArea';
import InstructionsArea from './InstructionsArea';
import EditVerseArea from './EditVerseArea';
import CommentArea from './CommentArea';
import style from '../css/Style';
// helpers
import * as checkAreaHelpers from '../helpers/checkAreaHelpers';

class CheckArea extends Component {
  render() {
    const {
      actions,
      contextIdReducer: { contextId },
      mode,
      tags,
      verseText,
      verseChanged,
      comment,
      selectionsReducer,
      projectDetailsReducer,
      resourcesReducer: { bibles },
      toolsReducer
    } = this.props;

    let alignedGLText = contextId.quote;
    const selectedGL = projectDetailsReducer.currentProjectToolsSelectedGL[toolsReducer.currentToolName];
    if (bibles[selectedGL] && bibles[selectedGL]['ult']) {
      const verseObjects = bibles[selectedGL]['ult'][contextId.reference.chapter][contextId.reference.verse].verseObjects;
      const wordsToMatch = contextId.quote.split(' ');
      const text = checkAreaHelpers.getAlignedText(verseObjects, wordsToMatch, contextId.occurrence);
      if (text) {
        alignedGLText = text;
      }
    }

    let modeArea;
    switch (mode) {
      case 'edit':
        modeArea = (
          <EditVerseArea
            tags={tags}
            verseText={verseText}
            verseChanged={verseChanged}
            actions={actions}
            dir={projectDetailsReducer.manifest.target_language.direction}
          />
        );
        break;
      case 'comment':
        modeArea = <CommentArea comment={comment} actions={actions} />;
        break;
      case 'select':
        modeArea = (
          <div style={{ WebkitUserSelect: 'none', display: "flex", flex: "1", justifyContent: "center", alignItems: "center", overflow: "auto" }}>
            <InstructionsArea
              verseText={verseText}
              selectionsReducer={selectionsReducer}
              alignedGLText={alignedGLText}
              mode={mode}
            />
          </div>);
        break;
      case 'default':
      default:
        modeArea = (
          <div style={{ WebkitUserSelect: 'none', display: "flex", justifyContent: "center", alignItems: "center", height: "100%" }}>
            <InstructionsArea
              dontShowTranslation={true}
              verseText={verseText}
              selectionsReducer={selectionsReducer}
              alignedGLText={alignedGLText}
            />
          </div>
        );
    }

    return (
      <div style={style.checkArea}>
        {mode === 'select' ?
          <SelectionArea
            {...this.props}
            actions={actions}
          /> :
          <DefaultArea {...this.props} />}
        <div style={{ borderLeft: '1px solid var(--border-color)', flex: 1, overflowY: "auto", display:'flex', justifyContent:'center' }}>
          {modeArea}
        </div>
      </div>
    );
  }
}

CheckArea.propTypes = {
  actions: PropTypes.object.isRequired,
  mode: PropTypes.string.isRequired,
  tags: PropTypes.array.isRequired,
  verseText: PropTypes.string.isRequired,
  verseChanged: PropTypes.bool.isRequired,
  comment: PropTypes.string.isRequired,
  contextIdReducer: PropTypes.shape({
    contextId: PropTypes.object
  }).isRequired,
  selectionsReducer: PropTypes.shape({
    selections: PropTypes.array
  }).isRequired,
  projectDetailsReducer: PropTypes.shape({
    manifest: PropTypes.object,
    currentProjectToolsSelectedGL: PropTypes.object
  }).isRequired,
  resourcesReducer: PropTypes.shape({
    bibles: PropTypes.object
  }).isRequired,
  toolsReducer: PropTypes.shape({
    currentToolName: PropTypes.string
  }).isRequired
};

export default CheckArea;
