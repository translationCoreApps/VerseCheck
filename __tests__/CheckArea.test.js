/* eslint-env jest */
import React from 'react';
import fs from 'fs-extra';
import { shallow } from 'enzyme';
import MuiThemeProvider from 'material-ui/styles/MuiThemeProvider';
import renderer from 'react-test-renderer';
import CheckArea from '../src/components/CheckArea';

const ult_titus_1 = '__tests__/fixtures/ult/tit/1.json';

describe('CheckArea component Tests', () => {
  const titus1 = fs.readJSONSync(ult_titus_1);
  const mock_translate = (text) => { return text; };
  let props;
  
  beforeEach(()=>{
    props = {
      translate: mock_translate,
      actions: {
        handleGoToNext: () => jest.fn(),
        handleGoToPrevious: () => jest.fn(),
        handleOpenDialog: () => jest.fn(),
        handleCloseDialog: () => jest.fn(),
        skipToNext: () => jest.fn(),
        skipToPrevious: () => jest.fn(),
        changeSelectionsInLocalState: () => jest.fn(),
        changeMode: () => jest.fn(),
        handleComment: () => jest.fn(),
        checkComment: () => jest.fn(),
        cancelComment: () => jest.fn(),
        saveComment: () => jest.fn(),
        handleTagsCheckbox: () => jest.fn(),
        handleEditVerse: () => jest.fn(),
        checkVerse: () => jest.fn(),
        cancelEditVerse: () => jest.fn(),
        saveEditVerse: () => jest.fn(),
        validateSelections: () => jest.fn(),
        toggleReminder: () => jest.fn(),
        openAlertDialog: () => jest.fn(),
        selectModalTab: () => jest.fn()
      },
      mode: 'default',
      tags: [],
      verseText: 'This is the verse text',
      verseChanged: false,
      comment: 'Comment!',
      contextIdReducer: {
        contextId: {
          quote: 'Θεοῦ',
          occurrence: 2,
          occurrences: 2,
          reference: {
            chapter: 1,
            verse: 1
          }
        }
      },
      selectionsReducer: {
        selections: []
      },
      selections: [],
      projectDetailsReducer: {
        manifest: {
          target_language: {
            direction: 'ltr'
          },
          project: {
            name: 'Titus'
          }
        },
        currentProjectToolsSelectedGL: {
          translationWords: 'en'
        }
      },
      resourcesReducer: {
        bibles: {
          en: {
            ult: {
              1: {
                1: titus1[1]
              }
            }
          },
          targetLanguage: {
            targetBible: {}
          }
        }
      },
      toolsReducer: {
        currentToolName: 'translationWords'
      }
    };    
  });

  test('Check CheckArea component mode default', () => {
    const component = renderer.create(
      <MuiThemeProvider>
        <CheckArea {...props} />
      </MuiThemeProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Test CheckArea component mode select)', () => {
    props.mode = 'select';
    const component = renderer.create(
      <MuiThemeProvider>
        <CheckArea {...props} />
      </MuiThemeProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Test CheckArea component mode edit)', () => {
    props.mode = 'edit';
    const component = renderer.create(
      <MuiThemeProvider>
        <CheckArea {...props} />
      </MuiThemeProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Test CheckArea component mode comment)', () => {
    props.mode = 'comment';
    const component = renderer.create(
      <MuiThemeProvider>
        <CheckArea {...props} />
      </MuiThemeProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Test CheckArea component where GL Aligned Text not found (uses Greek quote instead)', () => {
    props.resourcesReducer.bibles.en.ult[1][1] = {};
    const component = renderer.create(
      <MuiThemeProvider>
        <CheckArea {...props} />
      </MuiThemeProvider>
    );
    expect(component.toJSON()).toMatchSnapshot();
  });

  test('Test getAlignedGLText() function in CheckArea Component', () => {
    const wrapper = shallow(<CheckArea {...props} />);
    const expectedAlignedGLText = 'God\'s';
    expect(wrapper.instance().getAlignedGLText()).toEqual(expectedAlignedGLText);
  });

  test('Test getAlignedGLText() function in CheckArea Component where the GL align is not found so uses the Greek', () => {
    const nonExistantQuote = 'does-not-exist';
    props.contextIdReducer.contextId.quote = nonExistantQuote;
    const wrapper = shallow(<CheckArea {...props} />);
    expect(wrapper.instance().getAlignedGLText()).toEqual(nonExistantQuote);
  });
});
