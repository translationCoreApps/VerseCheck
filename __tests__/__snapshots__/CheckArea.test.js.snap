// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckArea component Tests Check CheckArea component mode default 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid var(--border-color)",
      "display": "flex",
      "flex": "1 0 130px",
      "fontSize": "1.1em",
    }
  }
>
  <div
    style={
      Object {
        "WebkitUserSelect": "none",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": "0 0 45px",
          "justifyContent": "space-between",
          "margin": "5px 10px 5px 15px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <span
          style={
            Object {
              "color": "var(--text-color-light)",
            }
          }
        >
          Titus
           
          1:1
        </span>
      </div>
      <div
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-fullscreen"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Click to show expanded verses"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "direction": "ltr",
          "flex": "auto",
          "overflowY": "auto",
          "padding": "0 15px 10px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "var(--text-color-light)",
            "userSelect": "none",
          }
        }
      >
        <span>
          This is the verse text
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "borderLeft": "1px solid var(--border-color)",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "overflowY": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <div
        style={
          Object {
            "overflow": "auto",
            "padding": "5px",
            "textAlign": "center",
          }
        }
      >
        <span>
          No selection has been made.
        </span>
        <br />
        <span>
          Click the Select button, then select the translation for this check.
        </span>
        <br />
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckArea component Tests Test CheckArea component mode comment) 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid var(--border-color)",
      "display": "flex",
      "flex": "1 0 130px",
      "fontSize": "1.1em",
    }
  }
>
  <div
    style={
      Object {
        "WebkitUserSelect": "none",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": "0 0 45px",
          "justifyContent": "space-between",
          "margin": "5px 10px 5px 15px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <span
          style={
            Object {
              "color": "var(--text-color-light)",
            }
          }
        >
          Titus
           
          1:1
        </span>
      </div>
      <div
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-fullscreen"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Click to show expanded verses"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "direction": "ltr",
          "flex": "auto",
          "overflowY": "auto",
          "padding": "0 15px 10px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "var(--text-color-light)",
            "userSelect": "none",
          }
        }
      >
        <span>
          This is the verse text
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "borderLeft": "1px solid var(--border-color)",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "overflowY": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": "auto",
          "flexDirection": "column",
          "height": "100%",
          "padding": "5px 15px 0 15px",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        <span
          className="glyphicon glyphicon-comment"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Comment
      </div>
      <div
        className="form-group"
        style={
          Object {
            "display": "flex",
            "flex": "auto",
          }
        }
      >
        <textarea
          autoFocus={true}
          className="form-control"
          defaultValue="Comment!"
          id="formControlsTextarea"
          onBlur={[Function]}
          onInput={[Function]}
          style={
            Object {
              "flex": "auto",
            }
          }
          type="text"
        />
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckArea component Tests Test CheckArea component mode edit) 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid var(--border-color)",
      "display": "flex",
      "flex": "1 0 130px",
      "fontSize": "1.1em",
    }
  }
>
  <div
    style={
      Object {
        "WebkitUserSelect": "none",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": "0 0 45px",
          "justifyContent": "space-between",
          "margin": "5px 10px 5px 15px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <span
          style={
            Object {
              "color": "var(--text-color-light)",
            }
          }
        >
          Titus
           
          1:1
        </span>
      </div>
      <div
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-fullscreen"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Click to show expanded verses"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "direction": "ltr",
          "flex": "auto",
          "overflowY": "auto",
          "padding": "0 15px 10px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "var(--text-color-light)",
            "userSelect": "none",
          }
        }
      >
        <span>
          This is the verse text
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "borderLeft": "1px solid var(--border-color)",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "overflowY": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": "auto",
          "flexDirection": "column",
          "height": "100%",
          "padding": "5px 15px 0 15px",
        }
      }
    >
      <div
        style={
          Object {
            "fontWeight": "bold",
          }
        }
      >
        <span
          className="glyphicon glyphicon-pencil"
          style={
            Object {
              "marginRight": "5px",
            }
          }
        />
        Edit Verse
      </div>
      <div
        className="form-group"
        style={
          Object {
            "display": "flex",
            "flex": "auto",
            "flexDirection": "column",
            "marginBottom": "5px",
          }
        }
      >
        <textarea
          autoFocus={true}
          className="form-control"
          defaultValue="This is the verse text"
          id="formControlsTextarea"
          onBlur={[Function]}
          onInput={[Function]}
          style={
            Object {
              "direction": "ltr",
              "flex": "auto",
            }
          }
          type="text"
        />
        <div
          style={
            Object {
              "flex": "0 0 65px",
              "fontSize": "0.9em",
              "marginTop": "5px",
            }
          }
        >
          First, make changes to verse above
          <br />
          <div
            style={
              Object {
                "display": "flex",
              }
            }
          >
            <div
              style={
                Object {
                  "flex": "1",
                }
              }
            >
              <div
                className="checkbox disabled"
                style={
                  Object {
                    "color": "var(--text-color-light)",
                    "marginLeft": "10px",
                    "marginRight": "10px",
                  }
                }
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Spelling
                </label>
              </div>
              <div
                className="checkbox disabled"
                style={
                  Object {
                    "color": "var(--text-color-light)",
                    "marginLeft": "10px",
                    "marginRight": "10px",
                  }
                }
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Punctuation
                </label>
              </div>
              <div
                className="checkbox disabled"
                style={
                  Object {
                    "color": "var(--text-color-light)",
                    "marginLeft": "10px",
                    "marginRight": "10px",
                  }
                }
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Word Choice
                </label>
              </div>
            </div>
            <div
              style={
                Object {
                  "flex": "1",
                }
              }
            >
              <div
                className="checkbox disabled"
                style={
                  Object {
                    "color": "var(--text-color-light)",
                    "marginLeft": "10px",
                    "marginRight": "10px",
                  }
                }
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Meaning
                </label>
              </div>
              <div
                className="checkbox disabled"
                style={
                  Object {
                    "color": "var(--text-color-light)",
                    "marginLeft": "10px",
                    "marginRight": "10px",
                  }
                }
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Grammar
                </label>
              </div>
              <div
                className="checkbox disabled"
                style={
                  Object {
                    "color": "var(--text-color-light)",
                    "marginLeft": "10px",
                    "marginRight": "10px",
                  }
                }
              >
                <label
                  title=""
                >
                  <input
                    checked={false}
                    disabled={true}
                    onChange={[Function]}
                    type="checkbox"
                  />
                  Other
                </label>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckArea component Tests Test CheckArea component mode select) 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid var(--border-color)",
      "display": "flex",
      "flex": "1 0 130px",
      "fontSize": "1.1em",
    }
  }
>
  <div
    style={
      Object {
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": "0 0 45px",
          "justifyContent": "space-between",
          "margin": "5px 10px 5px 15px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <span
          style={
            Object {
              "color": "var(--text-color-light)",
            }
          }
        >
          Titus
           
          1:1
        </span>
      </div>
    </div>
    <div>
      <div
        style={
          Object {
            "direction": "ltr",
            "flex": "auto",
            "overflowY": "auto",
            "padding": "0 15px 10px",
          }
        }
      >
        <div
          onMouseEnter={[Function]}
          onMouseLeave={[Function]}
          onMouseUp={[Function]}
        >
          <span>
            This is the verse text
          </span>
        </div>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "borderLeft": "1px solid var(--border-color)",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "overflowY": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "display": "flex",
          "flex": "1",
          "justifyContent": "center",
          "overflow": "auto",
        }
      }
    >
      <div
        style={
          Object {
            "overflow": "auto",
            "padding": "5px",
            "textAlign": "center",
          }
        }
      >
        <span>
          Please select the translation for:
        </span>
        <br />
        <span>
          <strong
            style={
              Object {
                "color": "var(--accent-color)",
              }
            }
          >
            "God's"
          </strong>
        </span>
        <br />
      </div>
    </div>
  </div>
</div>
`;

exports[`CheckArea component Tests Test CheckArea component where GL Aligned Text not found (uses Greek quote instead) 1`] = `
<div
  style={
    Object {
      "borderBottom": "1px solid var(--border-color)",
      "display": "flex",
      "flex": "1 0 130px",
      "fontSize": "1.1em",
    }
  }
>
  <div
    style={
      Object {
        "WebkitUserSelect": "none",
        "display": "flex",
        "flex": 1,
        "flexDirection": "column",
      }
    }
  >
    <div
      style={
        Object {
          "display": "flex",
          "flex": "0 0 45px",
          "justifyContent": "space-between",
          "margin": "5px 10px 5px 15px",
        }
      }
    >
      <div
        style={
          Object {
            "display": "flex",
            "flexDirection": "column",
          }
        }
      >
        <span
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        />
        <span
          style={
            Object {
              "color": "var(--text-color-light)",
            }
          }
        >
          Titus
           
          1:1
        </span>
      </div>
      <div
        onClick={[Function]}
      >
        <span
          className="glyphicon glyphicon-fullscreen"
          style={
            Object {
              "cursor": "pointer",
            }
          }
          title="Click to show expanded verses"
        />
      </div>
    </div>
    <div
      style={
        Object {
          "direction": "ltr",
          "flex": "auto",
          "overflowY": "auto",
          "padding": "0 15px 10px",
        }
      }
    >
      <div
        style={
          Object {
            "color": "var(--text-color-light)",
            "userSelect": "none",
          }
        }
      >
        <span>
          This is the verse text
        </span>
      </div>
    </div>
  </div>
  <div
    style={
      Object {
        "borderLeft": "1px solid var(--border-color)",
        "display": "flex",
        "flex": 1,
        "justifyContent": "center",
        "overflowY": "auto",
      }
    }
  >
    <div
      style={
        Object {
          "WebkitUserSelect": "none",
          "alignItems": "center",
          "display": "flex",
          "height": "100%",
          "justifyContent": "center",
        }
      }
    >
      <div
        style={
          Object {
            "overflow": "auto",
            "padding": "5px",
            "textAlign": "center",
          }
        }
      >
        <span>
          No selection has been made.
        </span>
        <br />
        <span>
          Click the Select button, then select the translation for this check.
        </span>
        <br />
      </div>
    </div>
  </div>
</div>
`;
